gitlab.go

```diff
@@ -123,7 +123,10 @@ func NewClient(httpClient *http.Client, token string) *Client {
 	}
 
 	c := &Client{client: httpClient, token: token, UserAgent: userAgent}
-	c.SetBaseURL(defaultBaseURL)
+	if err := c.SetBaseURL(defaultBaseURL); err != nil {
+		// should never happen since defaultBaseURL is our constant
+		panic(err)
+	}
```

Comment:
Yeah... I guess the `panic` would be fine and likely even cleaner for users of the package. After we added the `panic` I will add a test for this function making sure it will never panic (unless someone altered the code and did not run the tests :wink: